cmd/jaeger/internal/extension/jaegerquery/server_test.go

```diff
@@ -0,0 +1,223 @@
+	return nil
+}
+
+func Test_NewSever(t *testing.T) {
+	telemetrySettings := component.TelemetrySettings{
+		Logger: zap.NewNop(),
+	}
+	server := newServer(createDefaultConfig().(*Config), telemetrySettings)
+	assert.NotNil(t, server)
+	assert.NotNil(t, server.config)
+	assert.NotNil(t, server.logger)
+}
+
+func Test_Dependencies(t *testing.T) {
+	expectedDependencies := []component.ID{component.NewID(component.Type("jaeger_storage"))}
+	telemetrySettings := component.TelemetrySettings{
+		Logger: zap.NewNop(),
+	}
+
+	server := newServer(createDefaultConfig().(*Config), telemetrySettings)
+	dependencies := server.Dependencies()
+
+	assert.Equal(t, expectedDependencies, dependencies)
+}
+
+func Test_Start(t *testing.T) {
+	expectedErrMessage := "cannot find primary storage : cannot find extension 'jaeger_storage' (make sure it's defined earlier in the config)"
+	telemetrySettings := component.TelemetrySettings{
+		Logger: zap.NewNop(),
+	}
+
+	server := newServer(createDefaultConfig().(*Config), telemetrySettings)
+	host := componenttest.NewNopHost()
+	err := server.Start(context.Background(), host)
+	assert.Equal(t, expectedErrMessage, err.Error())
+}
+
+func Test_StartStorageHost(t *testing.T) {
+	telemetrySettings := component.TelemetrySettings{
+		Logger: zap.NewNop(),
+	}
+
+	config := &Config{
+		TraceStorageArchive: "jaeger_storage",
+		TraceStoragePrimary: "jaeger_storage",
+	}
+	server := newServer(config, telemetrySettings)
+	host := storageHost{}
+	err := server.Start(context.Background(), host)
+	assert.Nil(t, err)
+}
+
+func Test_AddArchiveStorage(t *testing.T) {
+	host := componenttest.NewNopHost()
+
+	tests := []struct {
+		name           string
+		qSvcOpts       *querysvc.QueryServiceOptions
+		config         *Config
+		expectedOutput string
+		expectedErr    string
+	}{
+		{
+			name:           "Archive storage unset",
+			config:         &Config{},
+			qSvcOpts:       &querysvc.QueryServiceOptions{},
+			expectedOutput: `{"level":"info","msg":"Archive storage not configured"}` + "\n",
+			expectedErr:    "",
+		},
+		{
+			name: "Archive storage set",
+			config: &Config{
+				TraceStorageArchive: "random-value",
+			},
+			qSvcOpts:       &querysvc.QueryServiceOptions{},
+			expectedOutput: "",
+			expectedErr:    "cannot find archive storage factory: cannot find extension 'jaeger_storage' (make sure it's defined earlier in the config)",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			logger, buf := testutils.NewLogger()
+			telemetrySettings := component.TelemetrySettings{
+				Logger: logger,
+			}
+			server := newServer(tt.config, telemetrySettings)
+			err := server.addArchiveStorage(tt.qSvcOpts, host)
+			if tt.expectedErr == "" {
+				assert.NoError(t, err)
+			} else {
+				assert.Equal(t, tt.expectedErr, err.Error())
+			}
+
+			assert.Equal(t, tt.expectedOutput, buf.String())
+		})
+	}
+}
+
+func Test_MakeQueryOptions(t *testing.T) {
```

Comment:
Seems like you're going overboard with the tests - isn't this method going to be called anyway from some other tests? The two primary approaches for tests are:
  1. test all public methods
  2. add more tests as needed to improve code coverage

Reply:
Testing only the public methods makes sense. Thanks for sharing. I'll keep that in mind.